commit e9fd05c3e4f21a623c98feace601a02179522fea
Author: Anshuman Gupta <anshuman.gupta@intel.com>
Date:   Mon Jan 11 13:41:18 2021 +0530

    drm/i915/hdcp: Support for HDCP 2.2 MST shim callbacks
    
    Add support for HDCP 2.2 DP MST shim callback.
    This adds existing DP HDCP shim callback for Link Authentication
    and Encryption and HDCP 2.2 stream encryption
    callback.
    
    v2:
    - Added a WARN_ON() instead of drm_err. [Uma]
    - Cosmetic changes. [Uma]
    v3:
    - 's/port_data/hdcp_port_data' [Ram]
    - skip redundant link check. [Ram]
    v4:
    - use pipe instead of port to access HDCP2_STREAM_STATUS
    
    Cc: Ramalingam C <ramalingam.c@intel.com>
    Reviewed-by: Uma Shankar <uma.shankar@intel.com>
    Reviewed-by: Ramalingam C <ramalingam.c@intel.com>
    Tested-by: Karthik B S <karthik.b.s@intel.com>
    Signed-off-by: Anshuman Gupta <anshuman.gupta@intel.com>
    Link: https://patchwork.freedesktop.org/patch/msgid/20210111081120.28417-18-anshuman.gupta@intel.com

diff --git a/drivers/gpu/drm/i915/display/intel_display_types.h b/drivers/gpu/drm/i915/display/intel_display_types.h
index d27c7732a42f..24792102bcf6 100644
--- a/drivers/gpu/drm/i915/display/intel_display_types.h
+++ b/drivers/gpu/drm/i915/display/intel_display_types.h
@@ -378,6 +378,10 @@ struct intel_hdcp_shim {
 	int (*config_stream_type)(struct intel_digital_port *dig_port,
 				  bool is_repeater, u8 type);
 
+	/* Enable/Disable HDCP 2.2 stream encryption on DP MST Transport Link */
+	int (*stream_2_2_encryption)(struct intel_connector *connector,
+				     bool enable);
+
 	/* HDCP2.2 Link Integrity Check */
 	int (*check_2_2_link)(struct intel_digital_port *dig_port,
 			      struct intel_connector *connector);